src/module_wrap.cc

```diff
@@ -361,7 +361,7 @@ void ModuleWrap::Evaluate(const FunctionCallbackInfo<Value>& args) {
   Local<Module> module = obj->module_.Get(isolate);
 
   ContextifyContext* contextify_context = obj->contextify_context_;
-  std::shared_ptr<MicrotaskQueue> microtask_queue;
+  MicrotaskQueue* microtask_queue = nullptr;
```

Comment:
Handing a raw pointer to the ModuleWrap is a bit concerning to me as it might not be aware of the expected lifetime. Perhaps it could be stored as a shared_ptr instead so this part can stay the same? As in still shared_ptr, just eliminating the wrap class?

I don't have a strong opinion either way, but exposing a raw pointer outside the class that owns it always makes me a bit nervous. 😅

Reply:
The shared_ptr here does not provide any additional protection as the owner of the microtask queue, the ContextifyContext, is exposed via a raw pointer already, but I can add it back if that makes the change seemingly less scary.

Reply:
Actually thinking about it again, I think making it a shared_ptr is more dangerous than passing a raw pointer here, it gave you the false idea that it's okay for the ownership of the microtask to be shared outside the contextify context - but if you enqueue a microtask that access the contextify context (e.g. by touching any of the global variables, including the JS builtins), it's unsafe. The microtask queue's validity actually depends on its owner, so its ownership should not be shared.

Reply:
Ah, fair point. 👍🏻